javascript/.gitignore

```diff
@@ -0,0 +1,3 @@
+npm-debug.log
+node_modules
+dmsnell-diff-match-patch-*.tgz
```

Comment:
this doesn't belong here

Reply:
It's there because `npm pack` which I use to verify the package contents makes a file matching that pattern, and I always forget to delete it before doing `git add .`

feel free to remove it before merging.